### DISCUSSION TITLE
por que isso aconteceu no meu git ??????????

### DISCUSSION THREAD

**Discussion by jeandossantos** [frontendbr/forum#2000] [Perguntas e Respostas] [answered]
Olá a quem está lendo,
tive um acidente trágico no meu projeto ao fazer o git push, mas não sei o porque!

Estava fazendo um projeto de tarefas para deixar no git. Fiz o backend e logo depois partir para o front.
fiz todo um esforço para deixa o projeto apresentável mesmo com minha habilidades limitadas.
Finalmente estava quase terminando o projeto e decidi fazer o commit e o push....
Logo depois formatei o pc, troquei o sistema de linux para o win10, instalei todos os programas que uso e finalmente decidi finalizar o projeto de tarefas, mas... instalei as depedencias do backend até aí ok, mas quando fui fazer no frontend deu ruim...
fui olhar os arquivo no vscode mas só tinha o arquivo "package-lock.json" na pasta de backend!  fui olhar o git e estava assim:

![merdafeita](https://user-images.githubusercontent.com/40217619/106358971-633d5800-6320-11eb-9f18-c5f82f7ab5dd.png)

o problema já foi feito, os arquivos já foram perdidos, mas gostaria de pelo menos saber o que causou para não aconteceu novamente.

desculpas se escolhi o assunto errado para a pergunta.

Agradeço desde já!

**Comment by leandroruel** [comment]
aparentemente o push nao foi feito, ou algum erro deu assim que você deu o comando `git push` ou um erro ocorreu enquanto o push estava em progresso. tem que ficar atento no seu terminal se alguma mensagem de erro aparece

**Comment by ghost** [comment]
Talvez seja na hora de empurrar  os arquivos,o comando foi feito errado.Tem o problema  no git push, mas também  um problema  na configuração  do repositório. Espero  ter ajudado. 😁😁

**Reply by jeandossantos** [reply]
Bem, eu não acho que tenho feito algo errado já que foi o primeiro commit, mas tá né. 
Depois dessa vou olhar umas 10 vezes depois de fazer commit kkk
valeu pena atenção.

**Reply by ghost** [reply]
Então  apaga a pasta  e refaz o git add ou cria uma nova branch e faz o deploy.

**Reply by jeandossantos** [reply]
Como já disse antes, formatei o pc e perdi o frontend já o backend ficou salvo normalmente. 
Comecei agora a refazer o frontend.

**Reply by jeandossantos** [reply]
Vou aproveitar e perguntar aqui...
peguei só  o backend que está pronto e criei o projeto vue no mesmo local, 
mas quando tento instalar depedencias no vuejs está criando outra node_modules.
por que isso ?

**Comment by giovannism20** [comment, marked as answer]
Não sei se entendi o problema mas a princípio é um _submodule_, ou seja, era um repositório git que tinha um remote configurado e o github faz isso para apontar apenas para o repositório de destino. Inclusive as vezes isso é usado para projetos que precisam de dependências internas.

**Reply by jeandossantos** [reply]
hm... eu até olhei essa coisa de submodule, mas não entendi nada. 
Tá dificil mesmo o meu projeto backend e front são diferentes, mas estão no mesmo nível.
Estou tentando dá o git add . e está dando um warning;

```
hint: You've added another git repository inside your current repository.
hint: Clones of the outer repository will not contain the contents of
hint: the embedded repository and will not know how to obtain it.
hint: If you meant to add a submodule, use:
hint:
hint:   git submodule add <url> frontend
hint:
hint: If you added this path by mistake, you can remove it from the
hint: index with:
hint:
hint:   git rm --cached frontend
hint:
hint: See "git help submodule" for more information.
```
tenho certeza que já fiz dessa forma antes, inclusive tenho projetos assim no meu git, mas agora não sei porque tá isso. 
notei que tem a pasta .git no projeto do vue também talvez é pela existencia dele ?

de qualquer forma, vou continuar tentando

**Reply by jeandossantos** [reply]
Sim, encontrei um artigo aqui antes e tentei pareceu dar certo por enquanto.

como fiz:

- excluí a pasta .git do projeto do vue (submodule)
- executei o comando no meu repositório git principal: 'git rm -f frontend'
- depois dei git add, commit e push normalmente e pareceu funcionar.

de qualquer forma, agradeço!

**Reply by giovannism20** [reply]
> hm... eu até olhei essa coisa de submodule, mas não entendi nada.
> Tá dificil mesmo o meu projeto backend e front são diferentes, mas estão no mesmo nível.
> Estou tentando dá o git add . e está dando um warning;
> 
> ```
> hint: You've added another git repository inside your current repository.
> hint: Clones of the outer repository will not contain the contents of
> hint: the embedded repository and will not know how to obtain it.
> hint: If you meant to add a submodule, use:
> hint:
> hint:   git submodule add <url> frontend
> hint:
> hint: If you added this path by mistake, you can remove it from the
> hint: index with:
> hint:
> hint:   git rm --cached frontend
> hint:
> hint: See "git help submodule" for more information.
> ```
> 
> tenho certeza que já fiz dessa forma antes, inclusive tenho projetos assim no meu git, mas agora não sei porque tá isso.
> notei que tem a pasta .git no projeto do vue também talvez é pela existencia dele ?
> 
> de qualquer forma, vou continuar tentando

Bom que o problema foi resolvido mas vou explicar sobre a tua pergunta da dúvida anterior. O fato de você ter múltiplas pastas dentro do teu projeto principal versionadas (.git) são lidas pelo versionador como subdomules onde o que ele vai fazer é apontar apenas pro repositório onde se encontra o código (caso este esteja no github).

@felipefialho Problema resolvido pode fechar a issue ;) :octocat:

**Reply by jeandossantos** [reply]
Acho que entendi agora sobre o submodule. 
obrigado pela atenção galera!
